### PR TITLE
Add automated checks for build scripts

### PR DESCRIPTION
**Pull request by miyuki**
This change:
- Adds a shell script to run precommit checks on Python scripts and
  mentions it in the README.md file
- Adds a Github workflow which runs the same checks on pull requests
  and updates on the 'main' branch

### CODE REVIEW

File: run-precommit-checks.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (vhscampos) [inline run-precommit-checks.sh:1]: Should we have this script portable instead?

Comment (vhscampos) [reply run-precommit-checks.sh:1]: OS-wise

Comment (miyuki) [reply run-precommit-checks.sh:1]: We are already using bash in setup.sh, so I guess it should be OK here as well.

**Review by vhscampos — APPROVED** [review]
LGTM

**Pull request merged**